### ISSUE TITLE
Request(s): We can Add Entry Lines, but not remove. Also an idea.

### ISSUE THREAD

**Issue by NaPurHab** [BiglySoftware/BiglyBT#1781]
**Request**
We can add entry lines in locations like `Interface: Tags: Files` and `Files: Torrents`. But there is no way to remove lines.

Could you add an Icon similar to the following at the End of each Line?
![image](https://user-images.githubusercontent.com/37715552/93398516-b6b77980-f849-11ea-9ade-58211b59cfa6.png)

**Idea**
The above was taken from the Queue of Handbrake.  The Queue window in Handbrake is usually hidden, and only shows when the user chooses to show it.  Just like the Options Window only shows when opened.  It might be worth considering doing this for all File operations. As an example, I had to move some torrents from one drive to another. While that was going, I found more to move, but could not start them, until the existing move was finished.  A file operations Queue would be useful.  It would Stack operations and do them in sequence.  It would also allow for canceling an operation that has not started. If I wanted to reorganize my storage, I could do a bunch of move operations and they would run one after the other. I could cancel any that had not started if I changed my mind.  The queue could also handle Copy, Checking and Allocating.  This would make all these Disk Intensive operations into a FIFO Queue. This would prevent Disk Thrashing, when Multiple Operations are taking place on one Drive.  It is possible to have Allocating, Checking and Moving happening at the same time on the same drive. I have seen this happen when Allocating and Checking happen at the same time. And it only gets slower when I do a move that reads or writes to the same drive.  Not only does BBT Uploads and Downloads slow to a halt. But each of the conflicting file intensive operations also get slower.
So, a File Operations Queue, inspired by the Queue in Handbrake.

_I know I can be a bit long winded, but it makes the idea clear in my own mind._

**Comment by parg** [comment]
I understand what you're requesting, something similar to the 'Transcode Queue' that BiglyBT currently has for devices.

Quite a bit of work though...

**Comment by NaPurHab** [comment]
Hmm, never did transcoding in BBT. I use Handbarke and XMediaRcode.   But FIFO on all heavy disk activities would help great. I just moved a 238gb torrent to another drive. I then found more to move but BBT was already busy with the move.  At another time I had a large allocate running and it was already slowing my uploads, then a move added to that slowed things even more.

As to the **Request**, I just thought of one restriction.  If trying to delete the last line, it would remove it, but add a new blank line.  There must always be one.

**Comment by NaPurHab** [comment]
I see in 2.5.0.1_B01-B02 that you have added Disk Operations View. But I can't find the View.  I had envisioned it as similar to how if I select several torrents and issue a Move it does it in a Queue fashion. The only thing needed is to expand that to include Move/Copy/Checking/Allocating, and have an additive function and a way to cancel an item that is waiting.

I did a test when B01 came out. I did a move, then found another torrent I wanted to move and started that. Both opened a separate dialog and ran slowly, and transfers dropped in speed until they finished.

So where is this View?

**Comment by parg** [comment]
Under Devices in the sidebar

**Comment by NaPurHab** [comment]
There is no Devices on my sidebar.

![image](https://user-images.githubusercontent.com/37715552/94246699-14069700-feea-11ea-9ad4-a78bfc14cb26.png)

**Comment by parg** [comment]
Maybe you removed devices in the options? 

Tools->Options->Interface->Display: Side Bar: Header order (number 5)

**Comment by parg** [comment]
If you want it always visible then right-click it in the sidebar and select 'add to->sidebar' (or maybe quick links)

**Comment by NaPurHab** [comment]
Ok, tried the add to side bar. Looks weird and narrow.  But I can't remove it from the sidebar now.

Also, under Devices, looks good, But a column header for File Size would be nice, to know when a Big torrent is taking a long time. And an alternate choice for progress as TEXT, 'Allocating  99.9%' or 'Moving 99.9%', etc.  It would feel more like when it is in the Library View, and the Status show Allocating % or Moving %.

**Comment by NaPurHab** [comment]
![image](https://user-images.githubusercontent.com/37715552/94249687-3e5a5380-feee-11ea-8cf8-a45964c3dc4f.png)

How do I remove an Add to Sidebar?

**Comment by NaPurHab** [comment]
What about the Request?

**Request**
We can add entry lines in locations like `Interface: Tags: Files` and `Files: Torrents`. But there is no way to remove lines.

Could you add an Icon similar to the following at the End of each Line?
![image](https://user-images.githubusercontent.com/37715552/93398516-b6b77980-f849-11ea-9ade-58211b59cfa6.png)

**Comment by parg** [comment]
Right-click on the header and select remove. Or go to the Dashboard entry, right click and select 'reset sidebar'

What about you having some patience?

**Comment by NaPurHab** [comment]
I had already tried the Right Click on Header. Tried Right Click + CTRL and Shift even the WinKey. No combo of clicking on the header works.  The Reset in Dashboad worked. But I never use Dashboard, so had no idea there was a reset in it.

As to patience, having quick responses because we are both awake. Makes it almost like Chat.
Since there are only 3 columns, it might work better to popup as a dialog, like when moving or copying.  It would just have more lines.

Anyway, G'nite, I'm off to bed.

**Comment by parg** [comment]
Pop up dialog is pain in arse to implement within existing frameworks.

**Comment by NaPurHab** [comment]
Well something to consider.   I just did a move to load balance disk activities.  Then I found another.  Here is what it looked like.  Note transfer speed has dropped to zero. This whole Idea was a FIFO Queue. So the 2nd move would queue.  Perhaps remove the Move Dialog and make it work through the Disk Operations.  And still needs a way to cancel an activity that is queued before it starts.  Right Click just has Copy to Clipboard???

![image](https://user-images.githubusercontent.com/37715552/94297555-af235f00-ff32-11ea-9b17-64198f8724b6.png)
